### ISSUE TITLE
Why is the "endpointController" enabled by default, but not pod/namespace controllers?

### ISSUE THREAD

**Issue by jayunit100** [miekg/coredns#5035]
This is a bit of a n00b question.  When you see something like `plugin/ready: Still waiting on: "kubernetes"` how do you know if its a `routing` issue, or an `rbac` issue, or both? 

Related: 

What controllers are generally considered "required" for a conformant, functioning Kubernetes cluster that uses coredns for its pod dns provider?

I tried to print out the decision making logic in the HasSynced() function, and i found something like:

```
[INFO] plugin/ready: Still waiting on: "kubernetes"
[INFO] plugin/kubernetes: dns ep controller is NOT NIL
[INFO] plugin/kubernetes: svc controller: false
[INFO] plugin/kubernetes: ep controller: false
[INFO] plugin/kubernetes: pod controller: true <-- this is not enabled though...
[INFO] plugin/kubernetes: ns controller: false
```

Which leads me to believe maybe we could add more fine grined debugging like ^ to give users a better idea of why the coredns pod isnt starting in certain scenarios (if this is true, im happy to file an issue to add granular logging to the HasSynced() function).

**Comment by chrisohaver** [comment]
> Why is the "endpointController" enabled by default, but not pod/namespace controllers?

Endpoints are required to answer queries for headless services, one of the requirements of the k8s dns spec. https://github.com/kubernetes/dns/blob/master/docs/specification.md

Pods that are not endpoints are not part of the spec, hence disabled by default.

Regarding debug. The lack of messaging in the log is due to a recent-ish change related to endpointslice version checking. When stuck at this point waiting for the api connection to work, it logs nothing. 
We could make the kubernetes plugin be verbose when trying to connect to the api at that stage. I actually started to add this recently, I’ll see if I can finish it off this week.

**Comment by jayunit100** [comment]
Thanks chris !
